StaticText/EnrichedString: Styling support

SmallJoker opened the pull request:
`StaticText` was not capable of styling, thus `GUIButton` does not work either any more (since 9284313).
This PR adds styling support to `EnrichedString` and makes use of its properties in `StaticText`.

Changes in `EnrichedString`:
 * "Override color" (from Styling) is used as default color until the color changes (escapes)
 * The length of the default color is kept and extended on concat operations
   * There's yet no "reset color" escape sequence, thus a single integer is enough for now.

Changes in `StaticText`:
 * Get and set color properties directly using `EnrichedString`
   * "duplicated" variables were removed
 * Unused features and broken features (bgcolor, override bgcolor) trimmed/simplified
 * Remove duplicated text due to `WordWrap`. Text format unified in `updateText()`

## To do

This PR is Ready for Review.

- [x] Testing using the minimal `/formspec` "unittest"
- [x] Testing using a custom formspec (textareas, labels, buttons)
- [x] Testing using craftguide
- [ ] ?? find more useful test formspecs

## How to test

1) Get an older Minetest build (5.1.0, or before formspec_version 3)
2) Get craftguide, minimal game, custom testing formspecs
3) Open the formspecs
4) Compare PR/old -> buttons and colors must look the same or more correct

SmallJoker committed:
StaticText/EnrichedString: Styling support

kilbith commented:
It's worth noting that the default background color of the tooltip changed from opaque green to translucent white, which isn't bad IMO.

Otherwise works :+1:

SmallJoker committed:
Fix tooltip fg/bgcolor

SmallJoker committed:
Fix default color for substr(), add unittests

SmallJoker commented:
@kilbith Thank you for this report. It turned out that `tooltip[` was broken because `setStaticText` cleared the override settings. Fixed that by converting the tooltip string to an `EnrichedString` directly.

Tested tooltips with `freetype = true` (not shown)
 * Foreground: yellow - OK
 * Background: red - OK
And `freetype = false` (see image below)
 * Foreground: not colorized (unsupported) OK
 * Background: red - OK

![grafik](https://user-images.githubusercontent.com/1497498/70860590-e1c01600-1f23-11ea-8417-4ed0f7b06667.png)

Styling also works (freetype enabled)
![grafik](https://user-images.githubusercontent.com/1497498/70860634-5430f600-1f24-11ea-96fd-2cdb9378d211.png)

For additional safety I added a new unittest function for EnrichedString (passed).

sfan5 reviewed (approved):
Works, code looks okay too
:+1: for unittests
